### PR TITLE
Proof hints for side conditions

### LINKED ISSUE #881: Emit side condition events into proof traces

### PR DESCRIPTION
**Pull request by gtrepta**
Fixes #881

### CODE REVIEW

File: lib/codegen/ProofEvent.cpp [hunk 1]
```diff
@@ -283,6 +284,42 @@ ProofEvent::functionEvent_post(llvm::BasicBlock *current_block) {
...
   llvm::BranchInst::Create(merge_block, true_block);
+
+  return merge_block;
+}
+
+llvm::BasicBlock *ProofEvent::scEvent(
+    KOREAxiomDeclaration *axiom, std::vector<llvm::Value *> const &args,
+    llvm::BasicBlock *current_block) {
+  auto [true_block, merge_block, outputFile]
+      = eventPrelude("side_condition", current_block);
```
Comment (Baltoli) [inline lib/codegen/ProofEvent.cpp:295]: This should check the global flag to see if proof hints are enabled or not.

File: include/kllvm/codegen/ProofEvent.h [hunk 2]
```diff
@@ -119,6 +120,10 @@ class ProofEvent {
   [[nodiscard]] llvm::BasicBlock *
   functionEvent_post(llvm::BasicBlock *current_block);
 
+  [[nodiscard]] llvm::BasicBlock *scEvent(
```
Comment (Baltoli) [inline include/kllvm/codegen/ProofEvent.h:123]: I have a mild preference for:
```suggestion
  [[nodiscard]] llvm::BasicBlock *sideConditionEvent(
```

File: docs/proof-trace.md [hunk 3]
```diff
@@ -42,7 +43,9 @@ hook        ::= WORD(0xAA) name arg* WORD(0xBB) kore_term
 ordinal     ::= uint64
 arity       ::= uint64
 variable    ::= name kore_term WORD(0xCC)
-rule        ::= ordinal arity variable*
+rule        ::= WORD(0x22) ordinal arity variable*
+
+side_cond   ::= WORD(0xEE) ordinal arity variable*
```
Comment (Baltoli) [inline docs/proof-trace.md:48]: I confirm that this matches the behaviour of the parser, and required minimal code changes on my end to get it working.

**Review by Baltoli — APPROVED** [review]
A couple of tiny changes, but feel free to merge when addressed.

File: docs/proof-trace.md (hunk 3 above)
Comment (gtrepta) [reply docs/proof-trace.md:48]: I added `WORD(0x22)` to the rule event for debugging purposes, I can remove it if you think it's unnecessary.

Comment (Baltoli) [reply docs/proof-trace.md:48]: I actually prefer it that way for consistency; we might be able to slim the format down a bit at some point but let's think about that once we have CI and downstream testing set up.

**Pull request merged**